docs/how-to/internal/x11passthrough.rst

```diff
@@ -0,0 +1,270 @@
+network interface bridge called ``docker0`` is installed. This bridge serves
+to mediate communication among containers as well as between containers and the
+host. If the host would like to expose a network-based service to a container,
+it can bind to a port on ``docker0`` and containers can connect to it.
+
+In Docker Desktop, however, the Linux VM in which containers run complicates
+matters. Inside the Linux VM, it's largely a similar configuration with a
+network bridge but the host machine cannot directly interact with this bridge
+interface. Instead, the host's network interface is assigned an address on the
+bridge similar to how other containers are. In this way, containers can still
+connect to network-based services on the host but not through a shared network
+interface called ``docker0``.
+
+To help simplify the configurations for applications running inside Docker
+Desktop containers, the hostname ``host.docker.internal`` will always resolve
+to an IP address for the host's network interface and thereby allow access to
+network-based services on the host.
+
+Unlike Docker Desktop, Docker Engine cannot intercept DNS requests from
+containers; therefore, ``host.docker.internal`` must be configured when the
+container is started. This is accomplished via the ``--add-host`` option which
+allows mapping a hostname to an arbitrary address for the hostname. This
+mapping is applied by writing it in the container's ``/etc/hosts`` file. Using
+``--add-host``, ``host.docker.internal`` is mapped to the keyword
+``host-gateway``. This keyword is a special value that the Docker server will
+replace with an address from which the host will be reachable within a
+container whether it is Docker Engine or Docker Desktop starting it.
+
+In conclusion, we can add ``--add-host host.docker.internal:host-gateway`` to
+the options to start a container and the host network interface will be
+reachable at ``host.docker.internal``.
+
+Exposing an X Display to a Container
+------------------------------------
+
+Given the knowledge of the operation of the Docker implementations, we finally
+have the pieces to expose an X display to a container. Since it is not possible
+to expose the display's socket directly to a container, a TCP proxy is
+configured to pass X messages on the network from the container to the socket
+on the host machine for the display.
+
+TCP Proxy
+~~~~~~~~~
+
+The `socat <https://repo.or.cz/socat.git>`__ tool is a widely available program
+to relay bi-directional data transfer between independent data channels. It
+allows running a process on the host to listen on a network port and send any
+received data to a socket connected to an X display on the other side.
+
+Creating a TCP proxy for the X display effectively creates a spoofed X display.
+The proxy is configured to listen on the TCP port for an unallocated display;
+the port number will be 6000 + the number of the display. Additionally, the
```

Comment:
Is this a standard, a convention, or something specific to Briefcase?

Reply:
Defined in the X11 standard.

https://www.x.org/releases/X11R7.7/doc/man/man1/Xorg.1.xhtml#heading5